www/src/components/withUrlSync.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import qs from 'qs';
+
+const updateAfter = 700;
+const searchStateToQueryString = searchState => ({
+  q: searchState.query,
+  p: searchState.page,
+  ...(searchState.refinementList && {
+    ...(searchState.refinementList['owner.name'] && {
+      owner: searchState.refinementList['owner.name'],
+    }),
+    ...(searchState.refinementList.keywords && {
+      keywords: searchState.refinementList.keywords,
+    }),
+  }),
+});
+
+const searchStateToUrl = searchState =>
+  searchState
+    ? `${window.i18n.url_base}/packages?${qs.stringify(
+        searchStateToQueryString(searchState)
+      )}`
+    : '';
+
+const queryStringToSearchState = queryString => {
+  const { p, q, owner, keywords } = qs.parse(queryString);
+  return {
+    query: q,
+    page: p || 1,
+    refinementList: {
+      ...(keywords && { keywords }),
+      ...(owner && { 'owner.name': owner }),
+    },
+  };
+};
+
+const originalPathName = location.pathname;
```

Comment:
this is going to be evaluated outside of the component lifecycle, which is bad. try evaluating it in a constructor

Reply:
ignore if not using file